crates/bevy_core/src/serde.rs

```diff
@@ -38,4 +40,41 @@ impl<'de> Visitor<'de> for EntityVisitor {
     fn visit_string<E: Error>(self, v: String) -> Result<Self::Value, E> {
         Ok(Name::new(v))
     }
+
+    /// We accidentally forgot to derive `ReflectSerialize` and `ReflectDeserialize` on `Name` for a while,
+    /// which meant that `Name` components were being serialized as `{ "hash": _, "name" _ }` maps.
+    ///
+    /// We've since fixed this on [#11447](https://github.com/bevyengine/bevy/pull/11447), but in order
+    /// to maintain backwards compatibility with data that might have been serialized with the wrong format,
+    /// we should keep support for deserializing it for a bit. (A couple of releases should be enough.)
+    fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
```

Comment:
Can we mark trait methods as deprecated?

If not, can we call a function that's marked deprecated so we can CTRL+F to find it?

Reply:
No 😕 I tried it but the compiler complained about that.

Let me try this calling other function hack.

Reply:
Okay, actually calling it was causing us to emit a warning on every build, which wasn't nice. I just left it around (starting with a `_` to prevent dead code warnings) for searchability.